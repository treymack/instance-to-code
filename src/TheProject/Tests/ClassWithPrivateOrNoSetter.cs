using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ClassWithPrivateOrNoSetter
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new GetterClass
                               {
                                   IntProp = 1,
                               };

            var code = ITC.Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }

    public class GetterClass
    {
        public GetterClass()
        {
            PrivateSetterIntProp = 6;
        }
        public int IntProp { get; set; }
        public int NoSetterIntProp { get { return 4; } }
        public int PrivateSetterIntProp { get; private set; }
    }
}