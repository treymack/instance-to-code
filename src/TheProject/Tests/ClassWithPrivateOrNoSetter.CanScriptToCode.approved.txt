new TheProject.Tests.GetterClass()
{
IntProp = 1,
}